test/gtest/common/test.h

```diff
@@ -32,62 +32,49 @@
 
 namespace ucs {
 
-/**
- * Base class for tests
- */
-class test_base {
-public:
-    typedef enum {
-        IGNORE_IF_NOT_EXIST,
-        FAIL_IF_NOT_EXIST,
-        SETENV_IF_NOT_EXIST,
-        SKIP_IF_NOT_EXIST
-    } modify_config_mode_t;
-
-    test_base();
-    virtual ~test_base();
-
-    void set_num_threads(unsigned num_threads);
-    unsigned num_threads() const;
-
-    virtual void set_config(const std::string& config_str = "");
-    virtual void modify_config(const std::string& name, const std::string& value,
-                               modify_config_mode_t mode = FAIL_IF_NOT_EXIST);
-    virtual void push_config();
-    virtual void pop_config();
+class log {
```

Comment:
- There are set of C-like callbacks which must be `static` anyway to avoid implicit `this` argument
 - `static log& instance()` makes usage a bit more complecated: `ucs::log::instance().num_errors()` instead of `ucs::log::num_errors()`
 - I implemented `namespace log` in first draft, IMO class is preferable in this case at least just because of incapsulation of all globas variable into single place

Reply:
1. all namespace functions are inherently "static", so you can pass them to C-like callbacks
2. ok
3. Well, you can still have all these variables declared in a single place, it will be just hidden inside the enum in cpp file. The advantage would be that those variables are not exposed outside